src/zcash/Proof.cpp

```diff
@@ -163,6 +163,10 @@ curve_G2 CompressedG2::to_libsnark_g2() const
 
     assert(r.is_well_formed());
 
+    if (alt_bn128_modulus_r * r != curve_G2::zero()) {
+        throw std::runtime_error("point is not in G2");
+    }
+
```

Comment:
I think so, though when/if libsnark changes those semantics we'll need to make sure we go through our codebase and understand the implications and assumptions broken by it.